### PR TITLE
Homepage (card-40 & card-41)

### PR DESCRIPTION
**Pull request by ezra-jbch**
Added cards for upcoming lessons in the home page plus more related to it.

THE BASE IS SET TO 'refactoring' BRANCH. CHANGE THE BASE TO MASTER IF IT NEED BE (if refactoring is already merged or will need further changes).

KNOWN ERROR: since there isn't a 53rd chapter information in the json files, it will cause an error for about a week at the end of 2023 and every 5-6 years afterwards.

This branch is branched off of refactoring branch. I broke nearly all good development practices and did it in all one in commit because I felt like I needed to get this done fast. I will make sure not to do it this way from now on. I did, however, tried to keep modularity in mind and did a fair bit of testing and styling.

Trello:

card-40: https://trello.com/c/lyHK8bNY
card-41: https://trello.com/c/gk1vAz7O

Added files:

- src/components/LessonCard.vue 
    reusable vue component that takes in group and either the date (today by default) or specific cycle and chapter as props to make a neat little card about that lesson

- src/components/UpcomingLessons.vue
    vue component that is placed in the home page to display info and lessonCard about lessons for next Sunday for each group
 
- src/lib/chronometry.js
    javascript file in lib folder for sharing functions related to time & date across the website for modularity & reusability

Modified files:

- src/data/constants.js
- src/views/Home.vue

### CODE REVIEW

File: src/lib/chronometry.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  return sundays; /*Array of sundays for the year*/
+};
+
+export function getChapterNum(d) {
+  // get the year of the upcoming Sunday of the given date 
+  const upcomingSunday = getNextSunday(d);
+  const sundaysOfYear = getSundaysByYear(upcomingSunday.getFullYear());
+  // check which number Sunday it is against the list of all Sundays that year
+  for (let i = 0; i < sundaysOfYear.length; i++) {
```
Comment (dlim716) [inline src/lib/chronometry.js:31]: Can be simplified to something like:
```
return sundaysOfYear.findIndex(sunday => sunday.getTime() === upcomingSunday.getTime() + 1;
```
You can read up on `findIndex` here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/findIndex

File: src/components/LessonCard.vue [hunk 2]
```diff
@@ -0,0 +1,196 @@
+<template>
```
Comment (dlim716) [inline src/components/LessonCard.vue:1]: Let's make sure we have trailing semicolons at the end of each line of code to be consistent in format.

File: src/components/LessonCard.vue [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+      // used only when isDateMode === true; given the date (default current Date), it automatically finds the upcoming cycle & chapter to use when displaying a lesson
+      date: {
+        type: Date,
+        default: new Date()
+      },
+    },
+
+    computed: {
+      groupTitle() {
+        return GROUP_TITLE_PER_ROUTE[this.groupRoute]
```
Comment (dlim716) [inline src/components/LessonCard.vue:99]: example of missing trailing semicolon.

File: src/components/UpcomingLessons.vue [hunk 4]
```diff
@@ -0,0 +1,69 @@
+<template>
```
Comment (dlim716) [inline src/components/UpcomingLessons.vue:1]: Let's make sure we have trailing semicolons at the end of each line of code to be consistent in format.

File: src/components/UpcomingLessons.vue [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import { getNextSunday, getChapterNum } from '../lib/chronometry.js'
+import LessonCard from './LessonCard.vue'
+
+export default {
+  components: { LessonCard },
+
+  data() {
+    return {
+      nextSunday: getNextSunday(new Date())
+    }
```
Comment (dlim716) [inline src/components/UpcomingLessons.vue:38]: example of missing trailing semicolon.

File: src/components/LessonCard.vue (hunk 2 above)
Comment (ezra-jbch) [reply src/components/LessonCard.vue:1]: Done!

File: src/components/UpcomingLessons.vue (hunk 4 above)
Comment (ezra-jbch) [reply src/components/UpcomingLessons.vue:1]: Done!

File: src/lib/chronometry.js (hunk 1 above)
Comment (ezra-jbch) [reply src/lib/chronometry.js]: Brilliant! Change made!

**Pull request merged**